SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -70,35 +107,51 @@ function Invoke-lpass {
-    if ($InputObject) {
-        switch ($true) {
-            {$UseNative -and $IgnoreErrors}  {$result = $InputObject | & $lpassPath @Arguments 2>&1; break}
-            {$UseNative}                     {$result = $InputObject | & $lpassPath @Arguments; break}
-            {$UseWSL -and $IgnoreErrors}     {$result = $InputObject | & wsl $lpassPath @Arguments 2>&1; break}
-            {$UseWSL}                        {$result = $InputObject | & wsl $lpassPath @Arguments; break}
-        }
-    } else {
-        switch ($true) {
-            { $UseNative -and $IgnoreErrors }   {$result = & $lpassPath @Arguments 2>&1; break}
-            { $UseNative }                      {$result = & $lpassPath @Arguments; break}
-            { $UseWSL -and $IgnoreErrors }      {$result = & wsl $lpassPath @Arguments 2>&1; break}
-            { $UseWSL }                         {$result = & wsl $lpassPath @Arguments; break}
+    $Params = @{
+        InputObject = $InputObject
+        UseNative   = $UseNative
+        lpassPath   = $lpassPath
+        Arguments   = $Arguments
+    }
+
+    # If we do redirect on the command themselves, it doesn't work.
+    # Doing redirect on the commands wrapped in a function work.
+    
+    if ($Arguments.Count -gt 0) {
+        switch ($Arguments[0]) {
+            'login' {  
+                # We want the prompt always, so no redirect
+                $result = Invoke-lpassInternal @Params
+            }
+            'show' {
+                # We might want the prompt, but are not sure yet.
+                $result = Invoke-lpassInternal @Params 2>&1
+                # If we get the message stating we might be logged out, we reissue the command without redirect (for Prompt)
+                if ($result -is [System.Management.Automation.ErrorRecord] -and [String]$result -like $lpassMessage.LoggedOut) {
+                    $result2 = Invoke-lpassInternal @Params
+                    # If $result2 -eq $null, something will have been printed in the console (because we disabled the redirect)
+                    # We therefore want to keep the original $result "Logged out" so it is thrown later on
+                    # If not $null, we want to evaluate $result2 instead and discard $result
+                    if ($null -ne $result2) {$result = $result2}
+                }
+            }
+            Default {
+                # By default, we always redirect streams
+                $result = Invoke-lpassInternal @Params 2>&1
+            }
         }
     }
 
     if ($result -is [System.Management.Automation.ErrorRecord]) {
         switch -Wildcard ([string] $result) {
-            $lpassMessage.LoggedOut { throw [PasswordRequiredException] $lpassMessage.LoggedOut }
+            $lpassMessage.LoggedOut { throw [PasswordRequiredException] "$($lpassMessage.LoggedOut.TrimEnd("*")) Connect-LastPass" }
```

Comment:
That's what I intended previously when I modified the `$lpassMessage.LoggedOut` string withotu consideration for its use in the switch statement.

Reply:
Gotcha! I find the `trim('*')` kinda ugly though...plus that trim will have to be rerun every single time... I think maybe we can use a new constant variable for and I'll be a tad clean. That also gives us a chance to change the text altogether. I'm not the biggest fan of this part that says `Could not find decryption key`. Maybe say "It appears you're not logged in, try logging in with Connect-LastPass" or something like that.